### PR TITLE
Ria 3278 state payment pending

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3278

### Change description ###
RIA-3278_State_Payment_Pending

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealPaymentConfirmation.java [hunk 1]
```diff
@@ -43,6 +43,8 @@ public PostSubmitCallbackResponse handle(
             new PostSubmitCallbackResponse();
 
         final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+        final String feeString = "\n\n#### Fee\n";
+        final String paymentReferenceNumber = "\n\n#### Payment reference number\n";
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealPaymentConfirmation.java:47]: I think using this variable name could be misleading. Possibly look to use the same naming convention as you used above for `feeString` (i.e. paymentReferenceNumberString or paymentReferenceNumberTitle). Alternatively define a CONST field for these repeated String usages.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmation.java [hunk 2]
```diff
@@ -31,22 +34,39 @@ public PostSubmitCallbackResponse handle(
             throw new IllegalStateException("Cannot handle callback");
         }
 
+        final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
         PostSubmitCallbackResponse postSubmitResponse =
             new PostSubmitCallbackResponse();
 
+        String payForAppealOfflineOption = null;
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmation.java:41]: May be worth renaming this field to `paymentOption`

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmationTest.java [hunk 3]
```diff
@@ -90,6 +95,40 @@ public void should_return_out_of_time_confirmation_when_out_of_time() {
         );
     }
 
+    @Test
+    public void should_return_confirmation_for_pay_offline_by_card_EA() {
```
Comment (bradsorour) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmationTest.java:99]: Try keep all to lowercase (standard format) for all unit test names

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java [hunk 4]
```diff
@@ -59,6 +62,70 @@ public void should_make_feePayment_and_update_the_case() {
         });
     }
 
+    @Test
+    public void should_write_pending_for_PA_Offline_payment() {
```
Comment (bradsorour) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java:66]: same as above for consistency

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java [hunk 5]
```diff
@@ -59,6 +62,70 @@ public void should_make_feePayment_and_update_the_case() {
...
+            verify(expectedUpdatedCase, times(1))
+                .write(PAYMENT_OFFLINE_FOR_DISPLAY, "Pending");
+
+            reset(callback);
+            reset(feePayment);
+        });
+    }
+
+    @Test
+    public void should_write_pending_for_HU_Offline_payment() {
```
Comment (bradsorour) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java:98]: same as above for consistency

**Review by bradsorour — COMMENTED** [review]
Hey Lee, just some minor comments. 

I'd also consider adding a functional test the tests the pay later (without a hearing scenario).

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealPaymentConfirmation.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealPaymentConfirmation.java:47]: Renamed to feeLabel and paymentReferenceNumberLabel

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmation.java (hunk 2 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmation.java:41]: Renamed to paymentOption

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmationTest.java (hunk 3 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AppealSubmittedConfirmationTest.java:99]: renamed

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java (hunk 4 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandlerTest.java:66]: lowercase renamed

**Comment by Lee365** [comment]
> Hey Lee, just some minor comments.
> 
> I'd also consider adding a functional test the tests the pay later (without a hearing scenario).

Added functional test for without hearing

File: src/functionalTest/resources/scenarios/RIA-3278-appeal-fee-pay-later-EA-change-state-to-payment-pending-with-hearing.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        "replacements": {
+          "payForTheAppeal": "payLater",
+          "appealType": "refusalOfEu",
+          "decisionHearingFeeOption": "decisionWithHearing",
+          "homeOfficeDecisionDate": "{$TODAY-14}",
+          "feeCode": "FEE0238",
+          "feeDescription": "Appeal determined with a hearing",
+          "feeVersion": 2,
+          "feeAmount": 140.0,
+          "paymentStatus": "Payment due",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3278-appeal-fee-pay-later-EA-change-state-to-payment-pending-with-hearing.json:21]: I think these fields need to be removed from the functional test in order to see if you're getting back the correct response.

Comment (Lee365) [reply src/functionalTest/resources/scenarios/RIA-3278-appeal-fee-pay-later-EA-change-state-to-payment-pending-with-hearing.json:21]: Yes - the fee details are not required. However, to change the appeal state to Appeal_Submitted, it requires the payment status of "Paid"

**Review by bradsorour — APPROVED** [review]
LGTM

**Review by mkayad — APPROVED** [review]
LGTM

**Review by mkayad — APPROVED** [review]
LGTM

**Review by mkayad — DISMISSED** [review]
LGTM

**Review by mkayad — DISMISSED** [review]
LGTM

**Review by mkayad — DISMISSED** [review]
LGTM

**Review by mkayad — DISMISSED** [review]
LGTM

**Review by mkayad — APPROVED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**